src/services/identity/LaunchClient.ts

```diff
@@ -35,21 +40,130 @@ class LaunchClient {
+  ): Promise<CreateDomainAssociationCommandOutput> => {
+    if (this.isTestEnv()) {
+      return this.mockCreateDomainAssociationOutput(input)
+    }
+    const output = this.amplifyClient.send(
+      new CreateDomainAssociationCommand(input)
+    )
+    return output
+  }
 
   createGetDomainAssociationCommandInput = (
     appId: string,
-    domainName: string
-  ): GetDomainAssociationCommandInput => ({
+    domainName: string,
+    subDomainSettings: SubDomainSetting[]
+  ):
+    | GetDomainAssociationCommandInput
+    | MockGetDomainAssociationCommandInput => ({
     appId,
     domainName,
+    ...(this.isTestEnv() && { subDomainSettings }),
   })
 
   sendGetDomainAssociationCommand = (
-    input: GetDomainAssociationCommandInput
-  ): Promise<GetDomainAssociationCommandOutput> =>
-    this.amplifyClient.send(new GetDomainAssociationCommand(input))
+    input:
+      | GetDomainAssociationCommandInput
+      | MockGetDomainAssociationCommandInput
+  ): Promise<GetDomainAssociationCommandOutput> => {
+    const isMockInput = "subDomainSettings" in input
+    if (isMockInput) {
+      // handle mock input
+      return this.mockGetDomainAssociationOutput(input)
+    }
+    return this.amplifyClient.send(new GetDomainAssociationCommand(input))
+  }
+
+  /**
+   * The rate limit for Create Domain Association is 10 per hour.
+   * We want to limit interference with operations, as such we mock this call during development.
+   * @returns Mocked output for CreateDomainAssociationCommand
+   */
+  mockCreateDomainAssociationOutput = (
+    input: CreateDomainAssociationCommandInput
+  ): Promise<CreateDomainAssociationCommandOutput> => {
+    const mockResponse: CreateDomainAssociationCommandOutput = {
+      $metadata: {
+        httpStatusCode: 200,
+      },
+      domainAssociation: {
+        autoSubDomainCreationPatterns: [],
+        autoSubDomainIAMRole: undefined,
+        certificateVerificationDNSRecord: undefined,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "CREATING",
+        enableAutoSubDomain: false,
+        statusReason: undefined,
+        subDomains: undefined,
+      },
+    }
+
+    const subDomainSettingsList = input.subDomainSettings
+    if (!subDomainSettingsList || subDomainSettingsList.length === 0) {
+      return Promise.resolve(mockResponse)
+    }
+
+    const subDomains: SubDomain[] = this.getSubDomains(subDomainSettingsList)
+
+    // We know that domainAssociation is not undefined, so we can use the non-null assertion operator
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    mockResponse.domainAssociation!.subDomains = subDomains
+    return Promise.resolve(mockResponse)
+  }
+
+  private isTestEnv() {
+    return config.get("env") === "test" || config.get("env") === "dev"
```

Comment:
ideally we should be using enums for comparison like ` === NODE_ENVS.TEST`

Reply:
I am in favour of not using enums tho, in line with Chin's reasoning in the previous PRs. Shall we discuss offline about this?

Reply:
for this, i recommended moving it to a `utils/` directory as it seems common enough for other services to use apart from LaunchClient

Regarding enums, we can always use a simple object instead right?

```
const Roles = {
  Admin: "admin",
  Writer: "writer",
  Reader: "reader"
} as const;

// Convert object key in a type
type RoleKeys = typeof Roles[keyof typeof Roles]

declare function hasAccess(role: RoleKeys): void;
```
Ref: https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh

Reply:
oh yes, agreed! Since this file no longer checks the env var (it checks to a new var `mockAmplifyCreateDomainAssociationCalls`), this is beyond the scope of the pr, happy to tackle this in a separate issue instead! Feel free to resolve this if you have no other concerns!